<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Gepoliano Chaves" />


<title>Phylogenetic Analysis of Brazilian mtDNA Sequences</title>

<script src="site_libs/header-attrs-2.22/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Programa Acadêmico de Estágio e Intercâmbio (PAEI)</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-solid fa-circle-info"></span>
     
    Sobre nós
  </a>
</li>
<li>
  <a href="detalhes-do-programa.html">
    <span class="fa fa-duotone fa-address-card"></span>
     
    Inscrições
  </a>
</li>
<li>
  <a href="eventos-de-traducao.html">
    <span class="fa fa-solid fa-earth-americas"></span>
     
    Eventos de Tradução
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-solid fa-laptop-file"></span>
     
    Projetos (2024)
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="projetos-pesquisa-2024-ucsc.html">UNILAB/UCSC</a>
    </li>
  </ul>
</li>
<li>
  <a href="mentores.html">
    <span class="fa fa-light fa-person-chalkboard"></span>
     
    Mentores
  </a>
</li>
<li>
  <a href="PRaD.html">
    <span class="fa fa-solid fa-people-arrows"></span>
     
    PmD
  </a>
</li>
<li>
  <a href="redes-novacao-empreendedorismo.html">
    <span class="fa fa-duotone fa-people-group"></span>
     
    Inovação e Empreendedorismo
  </a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Phylogenetic Analysis of Brazilian mtDNA
Sequences</h1>
<h4 class="author">Gepoliano Chaves</h4>
<h4 class="date">May 23rd, 2024</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#introduction" id="toc-introduction">1. Introduction</a>
<ul>
<li><a href="#mitochondrial-genome-and-matrilinear-information"
id="toc-mitochondrial-genome-and-matrilinear-information">1.1.
Mitochondrial genome and matrilinear information</a></li>
<li><a href="#mitochondrial-haplogroups"
id="toc-mitochondrial-haplogroups">1.2. Mitochondrial
haplogroups</a></li>
<li><a href="#download-of-mitochondrial-sequences"
id="toc-download-of-mitochondrial-sequences">1.3. Download of
mitochondrial sequences</a></li>
</ul></li>
<li><a href="#computational-imputation-and-databases"
id="toc-computational-imputation-and-databases">2. Computational
Imputation and Databases</a></li>
<li><a href="#alignment" id="toc-alignment">3. Alignment</a>
<ul>
<li><a href="#load-multi-fasta-mtdna-file"
id="toc-load-multi-fasta-mtdna-file">Load multi-fasta mtDNA
file</a></li>
<li><a href="#subset-from-positions-1bp-to-302bp"
id="toc-subset-from-positions-1bp-to-302bp">Subset from positions 1bp to
302bp</a></li>
</ul></li>
<li><a href="#alignment-visualization"
id="toc-alignment-visualization">3. Alignment Visualization</a>
<ul>
<li><a href="#load-library" id="toc-load-library">Load library</a></li>
<li><a href="#visualization-with-sample-names"
id="toc-visualization-with-sample-names">Visualization with sample
names</a></li>
<li><a href="#visualization-without-sample-names"
id="toc-visualization-without-sample-names">Visualization without sample
names</a></li>
</ul></li>
<li><a href="#visualize-haplogroup-classification-with-haplogrep-3"
id="toc-visualize-haplogroup-classification-with-haplogrep-3">5.
Visualize Haplogroup Classification with Haplogrep 3</a></li>
<li><a href="#distance-matrix-and-phylogenetic-tree-of-partial-sequence"
id="toc-distance-matrix-and-phylogenetic-tree-of-partial-sequence">6.
Distance Matrix and Phylogenetic Tree of Partial Sequence</a>
<ul>
<li><a href="#plot-distance-matrix" id="toc-plot-distance-matrix">Plot
distance matrix</a></li>
<li><a href="#construct-phylogenetic-trees"
id="toc-construct-phylogenetic-trees">Construct Phylogenetic
Trees</a></li>
<li><a href="#plot-phylogenetic-trees-extremo-sul"
id="toc-plot-phylogenetic-trees-extremo-sul">Plot Phylogenetic Trees
Extremo Sul</a></li>
<li><a href="#references" id="toc-references">7. References</a></li>
</ul></li>
</ul>
</div>

<div id="introduction" class="section level1">
<h1>1. Introduction</h1>
<div id="mitochondrial-genome-and-matrilinear-information"
class="section level2">
<h2>1.1. Mitochondrial genome and matrilinear information</h2>
<ul>
<li><p>Mitochondria is an organelle that contains the martrilineal
genetic information.</p></li>
<li><p>Mitochondrial genome is 16569 bp long (Bolze 2021) <span
class="citation">Bolze (2014)</span></p></li>
</ul>
<div class="float">
<img src="images/mt-genome.png" style="width:60.0%"
alt="Classification tree of mitochondrial haplogroups. Figure from Sabrina L. Mitchell (2014)." />
<div class="figcaption">Classification tree of mitochondrial
haplogroups. Figure from <span class="citation">Sabrina L. Mitchell
(2014)</span>.</div>
</div>
</div>
<div id="mitochondrial-haplogroups" class="section level2">
<h2>1.2. Mitochondrial haplogroups</h2>
<ul>
<li><p>Genetic variants (mutations) in mitochondrial DNA can be
organized in haplogroups.</p></li>
<li><p>Each matrilineal ancestry has characteristic
haplogroups.</p></li>
<li><p>This figure from Mitchell et al., 2014 shows the phylogenetic
tree of mitochondrial haplogroups in a large population of the
USA.</p></li>
</ul>
<div class="float">
<img src="images/mt-haplogroups.png" style="width:60.0%"
alt="Classification tree of mitochondrial haplogroups. Figure from Sabrina L. Mitchell (2014)." />
<div class="figcaption">Classification tree of mitochondrial
haplogroups. Figure from <span class="citation">Sabrina L. Mitchell
(2014)</span>.</div>
</div>
</div>
<div id="download-of-mitochondrial-sequences" class="section level2">
<h2>1.3. Download of mitochondrial sequences</h2>
<ul>
<li><p>Sequences of mitochondrion from Brazilians were downloaded from
GenBank.</p></li>
<li><p>The code to download such sequences are in the link
below.</p></li>
</ul>
<p><a href="https://paei.netlify.app/download_sequences">Download of
mitochondrial sequences.</a></p>
</div>
</div>
<div id="computational-imputation-and-databases" class="section level1">
<h1>2. Computational Imputation and Databases</h1>
<ul>
<li><p>A review on genotype imputation was writen by <span
class="citation">Yun Li and Abecasis (2009)</span>. They even mention
using the HapMap project as panels for genotype inference. They do not
mention an imputation workflow or protocol for genetic variants of
Brazilians. An imputation workflow was provided by <span
class="citation">Kalle Pärn (2018)</span>.</p></li>
<li><p>I was able to find an R function for imputation in <a
href="https://search.r-project.org/CRAN/refmans/dotgen/html/imp.html"
class="uri">https://search.r-project.org/CRAN/refmans/dotgen/html/imp.html</a></p></li>
<li><p>The function above is in package <a
href="https://cran.r-project.org/web/packages/dotgen/index.html"
class="uri">https://cran.r-project.org/web/packages/dotgen/index.html</a></p></li>
</ul>
<div class="float">
<img src="images/imputac%CC%A7a%CC%83o.png" style="width:40.0%"
alt="Reference panels can be used to create populational references. With milions or billions of sequences deposited in a database, a genetic model can be created or imputed. Figure from tellmeGen (n.d.)." />
<div class="figcaption">Reference panels can be used to create
populational references. With milions or billions of sequences deposited
in a database, a genetic model can be created or imputed. Figure from
<span class="citation">tellmeGen (n.d.)</span>.</div>
</div>
</div>
<div id="alignment" class="section level1">
<h1>3. Alignment</h1>
<div id="load-multi-fasta-mtdna-file" class="section level2">
<h2>Load multi-fasta mtDNA file</h2>
<p>Load package msa: multiple sequence alignment</p>
<pre class="r"><code>library(&quot;msa&quot;)</code></pre>
<p>Load multi-fasta mtDNA from Brazil.</p>
<pre class="r"><code>mt_brazilian &lt;- readDNAStringSet(file=&quot;./data/mt_brazilian.fasta&quot;)</code></pre>
<pre><code>## Warning in .Call2(&quot;fasta_index&quot;, filexp_list, nrec, skip, seek.first.rec, :
## reading FASTA file ./data/mt_brazilian.fasta: ignored 425 invalid one-letter
## sequence codes</code></pre>
</div>
<div id="subset-from-positions-1bp-to-302bp" class="section level2">
<h2>Subset from positions 1bp to 302bp</h2>
<p>Information about subsetting strings was gained from Chapter 4:
Manipulating Sequences with Biostrings, found in <span
class="citation">Gatto (2023)</span>.</p>
<pre class="r"><code>seq_start_to_end &lt;- subseq(mt_brazilian, start = 100, end = 200)</code></pre>
<p>Print sequence</p>
<pre class="r"><code>print(seq_start_to_end)</code></pre>
<pre><code>## DNAStringSet object of length 17:
##      width seq                                              names               
##  [1]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243780
##  [2]   101 ATGAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243781
##  [3]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243782
##  [4]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243783
##  [5]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243784
##  ...   ... ...
## [13]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243792
## [14]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243793
## [15]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243794
## [16]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243795
## [17]   101 ATAAAAACCCAATCCACATCAAA...CAACTGCAACTCCAAAGCCACC AF243796</code></pre>
<p>Align sequences</p>
<pre class="r"><code>start_time &lt;- Sys.time()
alignment_start_to_end &lt;- msa(seq_start_to_end)</code></pre>
<pre><code>## use default substitution matrix</code></pre>
<pre class="r"><code>end_time &lt;- Sys.time()
end_time - start_time</code></pre>
<pre><code>## Time difference of 0.07284594 secs</code></pre>
<pre class="r"><code>alignment_start_to_end</code></pre>
<pre><code>## CLUSTAL 2.1  
## 
## Call:
##    msa(seq_start_to_end)
## 
## MsaDNAMultipleAlignment with 17 rows and 102 columns
##      aln                                                   names
##  [1] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243780
##  [2] ATGAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243781
##  [3] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243789
##  [4] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243786
##  [5] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243795
##  [6] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243796
##  [7] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243785
##  [8] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243787
##  [9] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243794
## [10] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243783
## [11] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243784
## [12] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243791
## [13] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTC-AAAGCCACCC AF243788
## [14] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243793
## [15] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243782
## [16] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- AF243792
## [17] ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAATTCCAAAGCCACC- AF243790
##  Con ATAAAAACCCAATCCACATCAAAAC...ATCAACTGCAACTCCAAAGCCACC- Consensus</code></pre>
</div>
</div>
<div id="alignment-visualization" class="section level1">
<h1>3. Alignment Visualization</h1>
<div id="load-library" class="section level3">
<h3>Load library</h3>
<ul>
<li>Visualization of the Spike Protein Region alignment was made
possible with library ggmsa and the work of <span class="citation">Zhou
(2022)</span>.</li>
</ul>
<pre class="r"><code>library(ggmsa)</code></pre>
<pre><code>## Registered S3 methods overwritten by &#39;ggalt&#39;:
##   method                  from   
##   grid.draw.absoluteGrob  ggplot2
##   grobHeight.absoluteGrob ggplot2
##   grobWidth.absoluteGrob  ggplot2
##   grobX.absoluteGrob      ggplot2
##   grobY.absoluteGrob      ggplot2</code></pre>
<pre><code>## Registered S3 method overwritten by &#39;ggtree&#39;:
##   method      from 
##   identify.gg ggfun</code></pre>
<pre><code>## ggmsa v1.8.0  Document: http://yulab-smu.top/ggmsa/
## 
## If you use ggmsa in published research, please cite:
## L Zhou, T Feng, S Xu, F Gao, TT Lam, Q Wang, T Wu, H Huang, L Zhan, L Li, Y Guan, Z Dai*, G Yu* ggmsa: a visual exploration tool for multiple sequence alignment and associated data. Briefings in Bioinformatics. DOI:10.1093/bib/bbac222</code></pre>
</div>
<div id="visualization-with-sample-names" class="section level3">
<h3>Visualization with sample names</h3>
<ul>
<li>Plot alignment at around position 23063bp, showing variant
N501Y</li>
</ul>
<pre class="r"><code>ggmsa(mt_brazilian, 
      start = 40, end = 80, 
      char_width = 0.5, 
      seq_name = T) + geom_seqlogo() + geom_msaBar()</code></pre>
<pre><code>## Warning in rbind(c(&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, :
## number of columns of result is not a multiple of vector length (arg 9)</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate
## system, which will replace the existing one.</code></pre>
<p><img src="07-Phylogenetic-Analyisis_mt_BR_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="visualization-without-sample-names" class="section level3">
<h3>Visualization without sample names</h3>
<ul>
<li>Plot without sequence names</li>
</ul>
<pre class="r"><code>ggmsa(mt_brazilian, 
      start = 40, end = 80, 
      char_width = 0.5, 
      seq_name = F) + geom_seqlogo() + geom_msaBar()</code></pre>
<pre><code>## Warning in rbind(c(&quot;T&quot;, &quot;A&quot;, &quot;T&quot;, &quot;T&quot;, &quot;G&quot;, &quot;A&quot;, &quot;C&quot;, &quot;T&quot;, &quot;C&quot;, &quot;A&quot;, &quot;C&quot;, :
## number of columns of result is not a multiple of vector length (arg 9)</code></pre>
<pre><code>## Coordinate system already present. Adding new coordinate
## system, which will replace the existing one.</code></pre>
<p><img src="07-Phylogenetic-Analyisis_mt_BR_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
</div>
<div id="visualize-haplogroup-classification-with-haplogrep-3"
class="section level1">
<h1>5. Visualize Haplogroup Classification with Haplogrep 3</h1>
<p>Visualization is available <a
href="https://haplogrep.readthedocs.io/en/latest/annotations/#clusters-and-population-frequencies"
class="uri">https://haplogrep.readthedocs.io/en/latest/annotations/#clusters-and-population-frequencies</a></p>
<p><a
href="/Users/gepolianochaves/Desktop/Gepoliano/DNA%20do%20Brasil/matrilineal%20sequences/data/haplogroups%206%20eur/haplogroups.html">Haplogroup
Classification with Haplogrep 3</a></p>
</div>
<div id="distance-matrix-and-phylogenetic-tree-of-partial-sequence"
class="section level1">
<h1>6. Distance Matrix and Phylogenetic Tree of Partial Sequence</h1>
<p>This is necessary so function dist.ml can take the DNA.bin object and
process the Distance Matrix. Function dist.ml() is from package
phangorn</p>
<pre class="r"><code>library(&quot;phangorn&quot;)</code></pre>
<pre><code>## Warning: package &#39;phangorn&#39; was built under R version 4.1.2</code></pre>
<pre><code>## Loading required package: ape</code></pre>
<pre><code>## Warning: package &#39;ape&#39; was built under R version 4.1.2</code></pre>
<pre><code>## 
## Attaching package: &#39;ape&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biostrings&#39;:
## 
##     complement</code></pre>
<pre class="r"><code>dm_extremo_sul_msa_partial &lt;- dist.ml(as.DNAbin(alignment_start_to_end))</code></pre>
<div id="plot-distance-matrix" class="section level2">
<h2>Plot distance matrix</h2>
<p>Function table.paint() is from package adegenet. Load library
adegenet:</p>
<pre class="r"><code>library(&quot;adegenet&quot;)</code></pre>
<pre><code>## Warning: package &#39;adegenet&#39; was built under R version 4.1.2</code></pre>
<pre><code>## Loading required package: ade4</code></pre>
<pre><code>## Warning: package &#39;ade4&#39; was built under R version 4.1.2</code></pre>
<pre><code>## 
## Attaching package: &#39;ade4&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:Biostrings&#39;:
## 
##     score</code></pre>
<pre><code>## The following object is masked from &#39;package:BiocGenerics&#39;:
## 
##     score</code></pre>
<pre><code>## 
##    /// adegenet 2.1.10 is loaded ////////////
## 
##    &gt; overview: &#39;?adegenet&#39;
##    &gt; tutorials/doc/questions: &#39;adegenetWeb()&#39; 
##    &gt; bug reports/feature requests: adegenetIssues()</code></pre>
<pre><code>## 
## Attaching package: &#39;adegenet&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:phangorn&#39;:
## 
##     AICc</code></pre>
<pre class="r"><code>dm_extremo_sul_msa_partial_df &lt;- as.data.frame(as.matrix(dm_extremo_sul_msa_partial))
table.paint(dm_extremo_sul_msa_partial_df, cleg=0, clabel.row=.5, clabel.col=.5)</code></pre>
<p><img src="07-Phylogenetic-Analyisis_mt_BR_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>
<div id="construct-phylogenetic-trees" class="section level2">
<h2>Construct Phylogenetic Trees</h2>
<p>Phylogenetic tree UPGMA method</p>
<pre class="r"><code>treeUPGMA_extremo_sul_partial &lt;- upgma(dm_extremo_sul_msa_partial_df)</code></pre>
</div>
<div id="plot-phylogenetic-trees-extremo-sul" class="section level2">
<h2>Plot Phylogenetic Trees Extremo Sul</h2>
<p>Plot UPGMA Extremo Sul</p>
<pre class="r"><code>plot(treeUPGMA_extremo_sul_partial, main=&quot;UPGMA&quot;, col=&quot;red&quot;)</code></pre>
<p><img src="07-Phylogenetic-Analyisis_mt_BR_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
</div>
<div id="references" class="section level2">
<h2>7. References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Bolze2021" class="csl-entry">
Bolze, Alexandre. 2014. <em>Analysis of Mitochondrial DNA Variants in
200,000 Individuals</em>. <a
href="https://cdn.sanity.io/files/g5irbagy/production/960b88f3fd681a6207e9d18b9973e99fa6df9e1f.pdf">https://cdn.sanity.io/files/g5irbagy/production/960b88f3fd681a6207e9d18b9973e99fa6df9e1f.pdf</a>.
</div>
<div id="ref-Gatto4" class="csl-entry">
Gatto, Laurent. 2023. <em>Chapter 4 Manipulating Sequences with
Biostrings</em>. <a
href="https://uclouvain-cbio.github.io/WSBIM1322/sec-biostrings.html">https://uclouvain-cbio.github.io/WSBIM1322/sec-biostrings.html</a>.
</div>
<div id="ref-Pärn" class="csl-entry">
Kalle Pärn, Javier Nunez Fontarnau, Marita A. Isokallio. 2018.
<em>Genotype Imputation Workflow V3.0 v.1</em>. <a
href="https://www.protocols.io/view/genotype-imputation-workflow-v3-0-e6nvw78dlmkj/v1?step=3">https://www.protocols.io/view/genotype-imputation-workflow-v3-0-e6nvw78dlmkj/v1?step=3</a>.
</div>
<div id="ref-Mitchell2014" class="csl-entry">
Sabrina L. Mitchell, Kristin Brown‐Gentry, Robert Goodloe. 2014.
<em>Characterization of Mitochondrial Haplogroups in a Large
Population‐based Sample from the United States</em>. <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4113317/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4113317/</a>.
</div>
<div id="ref-tellmeGen" class="csl-entry">
tellmeGen. n.d. <em>Genética Preditiva</em>. Accessed May 2, 2024. <a
href="https://www.tellmegen.com/pt/genetica-preditiva">https://www.tellmegen.com/pt/genetica-preditiva</a>.
</div>
<div id="ref-Li2009" class="csl-entry">
Yun Li, Serena Sanna, Cristen Willer, and Gonçalo Abecasis. 2009.
<em>Genotype Imputation</em>. <a
href="https://pubmed.ncbi.nlm.nih.gov/19715440/">https://pubmed.ncbi.nlm.nih.gov/19715440/</a>.
</div>
<div id="ref-Zhou" class="csl-entry">
Zhou, Lang. 2022. <em>Ggmsa:a Visual Exploration Tool for Multiple
Sequence Alignment and Associated Data</em>. <a
href="http://yulab-smu.top/ggmsa/index.html">http://yulab-smu.top/ggmsa/index.html</a>.
</div>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
